mb_netmgmt/netconf.py

```diff
@@ -71,24 +71,27 @@ def open_upstream(self):
         to = self.get_to()
         if not to:
             return
-        timeout = getattr(to, "timeout", None)
-        if timeout is None:
-            # Try to get from proxy dict if available
-            if hasattr(self, "proxy") and isinstance(self.proxy, dict):
-                timeout = self.proxy.get("timeout")
-        if timeout is None:
-            timeout = 60
-        else:
-            timeout = int(timeout)
-        self.manager = connect(
-            host=to.hostname,
-            port=to.port or PORT_NETCONF_DEFAULT,
-            username=to.username,
-            password=to.password,
-            key_filename=self.key_filename,
-            hostkey_verify=False,
-            timeout=timeout,
-        )
+            
+        # Build connection parameters
+        connect_params = {
+            'host': to.hostname,
+            'port': to.port or PORT_NETCONF_DEFAULT,
+            'username': to.username,
+            'password': to.password,
+            'key_filename': self.key_filename,
+            'hostkey_verify': False,
+        }
+        
+        # Get timeout from proxy configuration if explicitly specified
+        proxy = self.get_proxy_config()
+        if proxy and "timeout" in proxy:
+            try:
+                timeout = int(proxy["timeout"])
+                connect_params['timeout'] = timeout
+            except (ValueError, TypeError):
+                pass
+            
+        self.manager = connect(**connect_params)
```

Comment:
there is no need (or advantage) to pack all those parameters into a separate dict. i think it's simpler to have a `timeout` variable which is `None` by default and gets the appropriate value when it's configured